service/src/main/python/jcarbon/tensorflow/callbacks.py

```diff
@@ -125,6 +97,98 @@ def on_epoch_end(self, epoch, logs=None):
         self.reports[epoch] = self.last_report.to_frame().assign(
             epoch=epoch).set_index('epoch', append=True)
 
+class JCarbonExperimentCallback(JCarbonChunkingCallback):
+    def __init__(
+            self,
+            addr='localhost:8980',
+            period_ms=DEFAULT_PERIOD_MS,
+            signals=DEFAULT_SIGNALS,
+            chunking_period_sec=DEFAULT_PERIOD_SECS):
+        super().__init__(addr, period_ms, signals)
+        self.timestamps = {}
+        self.batch_timestamps = None
+
+    def on_epoch_begin(self, epoch, logs=None):
+        super().on_epoch_begin(epoch, logs)
+        self.curr_batch_timestamps = None
+    
+    def on_train_batch_begin(self, batch, logs=None):
+        self.batch_start = time.time()
+        super().on_train_batch_begin(batch, logs)
+        
+        
+    def on_train_batch_end(self, batch, logs=None):
+        curr = time.time()
+        super().on_train_batch_end(batch, logs)
+        self.batch_end = int((10**9 * curr))
```

Comment:
Why are you assigning to this field? Is there a benefit I don't see here?

Reply:
oh good point, I can just assign it to `curr` since im not using that in a different field